pkg/util/resource/scanner_darwin.go

```diff
@@ -0,0 +1,20 @@
+//go:build darwin
+
+package resource
+
+import (
+	"github.com/go-kit/log/level"
+
+	"github.com/cortexproject/cortex/pkg/util/log"
+)
+
+type noopScanner struct{}
+
+func (s *noopScanner) scan() (float64, error) {
+	return 0, nil
+}
```

Comment:
I think we can move `noopScanner` to `scanner.go` as it is not specific to darwin. And can be used probably in test.

Not blockign this PR